### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by jschueller**
Few more commits before the release including what can be included from the KL sensitivity branch:
- Matrix::resize
- check KL algo threshold argument 

Also applied spelling fixes patch from #2016

### CODE REVIEW

**Review by sofianehaddad — APPROVED** [review]

File: lib/src/Base/Optim/OptimizationProblemImplementation.cxx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //                                               -*- C++ -*-
 /**
- *  @brief OptimizationProblemImplementation allows to describe an optimization problem
+ *  @brief OptimizationProblemImplementation allows one to describe an optimization problem
  *
  *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
```
Comment (sofianehaddad) [inline lib/src/Base/Optim/OptimizationProblemImplementation.cxx:5]: Could you also update the headers please?

Comment (jschueller) [reply lib/src/Base/Optim/OptimizationProblemImplementation.cxx:5]: the text is the same in both

Comment (sofianehaddad) [reply lib/src/Base/Optim/OptimizationProblemImplementation.cxx:5]: I mean `2005-2021` --> `2005-2022`

Comment (jschueller) [reply lib/src/Base/Optim/OptimizationProblemImplementation.cxx:5]: yes, I'll do that in the release branch to avoid too many diffs here

Comment (sofianehaddad) [reply lib/src/Base/Optim/OptimizationProblemImplementation.cxx:5]: ok thanks

**Pull request merged**